### PR TITLE
Added back divine shield for resurrected monsters if previously had it

### PR DESCRIPTION
**Pull request by jperucha**
Moved resurrect and removeDivineShield responsibilites to its own entity instead of having to pass the entity as a param and operate them in a different class

### CODE REVIEW

**Comment by thechungster** [comment]
Is divine shield the only thing that gets reset?
We should probably generalize it for any ability in case future ones are affected

**Comment by jperucha** [comment]
The scope of this PR was intented to be fixing the divine shield problem with resurrection.
There are several ways of fixing it and I chose the simplest one with a boolean as I don't want to do my first PR in this repo a new map that contained the initial state of the buffs map to be able to revert to that state the monster that was resurrected.
I also checked that abilities like inspire are added back to the resurrected monster so I don't know to which extent a copy of the buff map to revert the monster to that initial state is needed

File: src/game_monster.ts [hunk 1]
```diff
@@ -465,6 +471,13 @@ export class GameMonster extends GameCard {
     return currentMelee;
   }
 
+  resurrect() {
+    this.health = 1;
+    this.armor = this.startingArmor;
+    if (this.hadDivineShield) { this.addAbility(Ability.DIVINE_SHIELD) }
```
Comment (thechungster) [inline src/game_monster.ts]: missing semi colon

Move this to it's ownline
```
if (this.hadDivineShield) { 
  this.addAbility(Ability.DIVINE_SHIELD);
}
```

Comment (jperucha) [reply src/game_monster.ts]: Done

**Review by thechungster — APPROVED** [review]
Thanks!

**Pull request merged**